### PR TITLE
Set hex dashboard link on Hubspot companies

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

We are adding links to hex dashboards manually to Hubspot companies, but the link is in a consistent format and can be assigned automatically.

## How did you test this change?

Will click test after this change goes outl.

## Are there any deployment considerations?

Confirm property gets set as expected.

### CODE REVIEW

**Review by Vadman97 — APPROVED** [review]
nice! thank you

File: backend/hubspot/hubspot.go
Comment (Vadman97) [inline backend/hubspot/hubspot.go:1]: on line 349, if hubspot created the company already, we might need to update to add the custom properties

**Review by jay-khatri — APPROVED** [review]

File: backend/hubspot/hubspot.go [hunk 1]
```diff
@@ -373,8 +366,41 @@ func (h *HubspotApi) CreateCompanyForWorkspaceImpl(ctx context.Context, workspac
...
+
+	if companyID, err = pollHubspot(func() (*int, error) {
+		id, err := h.getCompany(ctx, name)
+		if err != nil {
+			return nil, err
+		}
+
+		log.WithContext(ctx).
+			WithField("name", name).
+			Warnf("company already exists in Hubspot. updating")
```
Comment (Vadman97) [inline backend/hubspot/hubspot.go:385]: this is the expected flow so this should be an `Infof`

File: backend/hubspot/hubspot.go [hunk 2]
```diff
@@ -373,8 +366,41 @@ func (h *HubspotApi) CreateCompanyForWorkspaceImpl(ctx context.Context, workspac
...
+		}
+
+		return id, nil
+	}); companyID != nil {
+		return
+	}
+
+	log.WithContext(ctx).
+		WithField("name", name).
+		Warnf("failed to get client-side hubspot company. creating")
```
Comment (Vadman97) [inline backend/hubspot/hubspot.go:401]: good idea to add this!

File: backend/hubspot/hubspot.go [hunk 3]
```diff
@@ -373,8 +366,41 @@ func (h *HubspotApi) CreateCompanyForWorkspaceImpl(ctx context.Context, workspac
...
+		if id != nil {
+			_, err := h.hubspotClient.Companies().Update(*id, companyProperties)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		return id, nil
+	}); companyID != nil {
+		return
```
Comment (Vadman97) [inline backend/hubspot/hubspot.go:396]: sorry, should have been a bit more precise here. i think the new logic should go out here to happen once, when the company is found. the inline function above is used to actually find the company in a loop until it appears

Comment (ccschmitz) [reply backend/hubspot/hubspot.go:396]: Oh, right! My bad.

**Review by Vadman97 — APPROVED** [review]

**Comment by jay-khatri** [comment]
Thanks Chris!

**Pull request merged**